### PR TITLE
Allow (optional) port number in repository URL

### PR DESCRIPTION
**Pull request by kornrunner**
This allows us to create webhooks for URLs that have port numbers in them (specifically `ssh://` running under custom port).

I've added the test as well, but haven't run them, haven't set it up properly yet.

### CODE REVIEW

File: src/Packagist/WebBundle/Controller/ApiController.php [hunk 1]
```diff
@@ -111,7 +111,7 @@ public function updatePackageAction(Request $request)
         }
 
         if (isset($payload['repository']['url'])) { // github/gitlab/anything hook
-            $urlRegex = '{^(?:ssh://git@|https?://|git://|git@)?(?P<host>[a-z0-9.-]+)[:/](?P<path>[\w.-]+/[\w.-]+?)(?:\.git)?$}i';
+            $urlRegex = '{^(?:ssh://git@|https?://|git://|git@)?(?P<host>[a-z0-9.-]+):(?:[0-9]+)?/(?P<path>[\w.-]+/[\w.-]+?)(?:\.git)?$}i';
```
Comment (stof) [inline src/Packagist/WebBundle/Controller/ApiController.php]: this change breaks all URLs which don't have the port in them

**Comment by stof** [comment]
@Seldaek why don't we have Travis on the repo to run the testsuite ?

**Comment by Seldaek** [comment]
Just never bothered to set it up. But yeah that change is broken, you can run the tests without setting up the whole project though. At least if you use phpunit --filter to only run the tests you modified, this doesn't depend on anything being setup.

**Comment by kornrunner** [comment]
Sorry guys, it's my bad either way. I tried to run It should be ok now.

Travis would be great, I agree.

**Comment by Seldaek** [comment]
Thanks for updating, looks better now :)

**Comment by kornrunner** [comment]
Also, this one is somewhat related (though in composer, actually): https://github.com/composer/composer/pull/4530.
It'd be great if you could check it out. Thank you!

**Pull request merged**